### PR TITLE
gauges: Remove breath indication; optimize HP setting code

### PR DESCRIPTION
**Pull request by ClobberXD**
This optimisation was suggested by @ranfdev in https://github.com/MT-CTF/capturetheflag/issues/320#issuecomment-529213581

****

This PR removes the breath indication from gauges (I didn't even know this existed until I started working on this), and updates the gauge entity in an `on_player_hpchange` callback instead of within `on_step`.

While this works as expected, I did notice *once*, that the HP indication kept alternating between the player's actual HP and full (20) while player was being healed by regen. So I've added a couple of debug print statements in a separate commit. Once this PR is confirmed to be working properly, I'll remove the second commit before merge.

### CODE REVIEW

File: mods/pvp/gauges/init.lua [hunk 1]
```diff
@@ -33,13 +34,9 @@ function hp_bar:on_step(dtime)
...
 	end
 
-	local hp = wielder:get_hp()
-	local breath = wielder:get_breath()
-	self.object:set_properties({
-		textures = {
-			"health_" .. tostring(hp) .. ".png^breath_" .. tostring(breath) .. ".png",
-		},
-	})
+	print("\t hp_bar:set_hp(" .. hp .. ")")
```
Comment (Panquesito7) [inline mods/pvp/gauges/init.lua]:
```suggestion
	minetest.log("info", "\t hp_bar:set_hp(" .. hp .. ")")
```

File: mods/pvp/gauges/init.lua [hunk 2]
```diff
@@ -84,6 +74,33 @@ function gauges.check_gauges()
...
+		end
+	end)
 end
-minetest.after(2, gauges.check_gauges)
+
+minetest.register_on_player_hpchange(function(player, hp_change)
+	print("*** on_player_hpchange")
+	local ent = gauges.entities[player:get_player_name()]
+	if ent then
+		print("\tInvoking ent:set_hp with final HP of [" .. player:get_hp() .. " + " .. hp_change .. "]")
```
Comment (Panquesito7) [inline mods/pvp/gauges/init.lua]:
```suggestion
		minetest.log("info", "\tInvoking ent:set_hp with final HP of [" .. player:get_hp() .. " + " .. hp_change .. "]")
```

File: mods/pvp/gauges/init.lua [hunk 3]
```diff
@@ -84,6 +74,33 @@ function gauges.check_gauges()
...
+		ent:set_hp(player:get_hp() + hp_change)
+	end
+	print("**********************")
```
Comment (Panquesito7) [inline mods/pvp/gauges/init.lua]:
```suggestion
	minetest.log("info", "**********************")
```

Comment (ClobberXD) [reply mods/pvp/gauges/init.lua]: All of the `print` statements are testing code, and will be removed before merge. So... no, just no. ;)

Comment (Panquesito7) [reply mods/pvp/gauges/init.lua]: Sorry for so much `minetest.log` suggestions.
The same for the other reviews (on other PRs).

**Comment by ClobberXD** [comment]
Tested alongside `master` and #457. Benchmark results: https://gist.github.com/ClobberXD/0999b88a90a2eccddd17264cd626576c

**Comment by ClobberXD** [comment]
I'm trying to combine this with #457. WIP for a bit...

**Comment by ClobberXD** [comment]
For some weird reason, `obj:get_pos()` returns `nil`...

**Pull request closed without merging**